provider/provider.go

```diff
@@ -72,13 +75,14 @@ func Call(provider, specPath string) (string, error) {
 
 // Given a naked filename, returns a path to executable prefixed with DefaultPath
 // This is so that "./provider" will work as expected.
-func expandPath(provider string) string {
+func expandPath(provider string) (string, error) {
 	// Base returns just the last path segment.
 	// If it's different, that means it's a (rel or abs) path
-	if path.Base(provider) != provider {
-		return provider
+	if filepath.Base(provider) != provider {
+		return filepath.Abs(provider)
 	}
-	return path.Join(DefaultPath, provider)
+
+	return filepath.Join(DefaultPath, provider), nil
 }
 
 func getDefaultPath() string {
```

Comment:
I think we can get some code coverage for the Windows case by splitting the getDefaultPath() function up, something like this:
```
func getDefaultPath() string {
	if runtime.GOOS == "windows" {
		programFilesDir := os.Getenv("ProgramW6432")
		return getWindowsDefaultPath(programFilesDir)
	}
	return "/usr/local/lib/summon"
}

func getWindowsDefaultPath(programFilesDir string) string {
	// Use "Program Files" directory if it's configured, otherwise
	// use a hardcoded value we think should be right.
	programFilesDir := os.Getenv("ProgramW6432")
	if programFilesDir == "" {
		programFilesDir = path.Join("C:", "Program Files")
		programFilesDir = filepath.Join("C:", "Program Files")
	}

	return path.Join(programFilesDir, "Cyberark Conjur", "Summon", "Providers")
	return filepath.Join(programFilesDir, "Cyberark Conjur", "Summon", "Providers")
}
```

Reply:
@diverdane Can you elaborate on this? Afaict `filepath.Join` will still return OS-specific path separator.

Reply:
Oh, you are right! Looks like *everything* in this function is OS-specific, so I don't think there's any OS-independent testing we can do. And if we added mockable interfaces, we'd just be testing our mocks.